code/modules/vehicles/unmanned/unmanned_droid.dm

```diff
@@ -29,6 +29,33 @@
 		STOP_PROCESSING(SSslowprocess, src)
 		user.clear_fullscreen("machine", 5)
 
+/obj/vehicle/unmanned/droid/welder_act(mob/living/user, obj/item/I)
+	if(user.do_actions)
+		balloon_alert(user, "Already busy!")
+		return FALSE
+	if(obj_integrity >= max_integrity)
+		return TRUE
+	if(user.skills.getRating("engineer") < SKILL_ENGINEER_ENGI)
+		balloon_alert_to_viewers("[user] fumbles repairs", ignored_mobs = user)
+		balloon_alert(user, "You fumble repair")
+		var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
```

Comment:
useless var, don't think that add that much readabiliy

Reply:
it makes the line stupid long this is fine imo